guest/rust/examples/experimental/procedural_mesh/src/client.rs

```diff
@@ -0,0 +1,77 @@
+use ambient_api::client::experimental::*;
+use ambient_api::components::core::camera::aspect_ratio_from_window;
+use ambient_api::components::core::primitives::cube;
+use ambient_api::concepts::{make_perspective_infinite_reverse_camera, make_transformable};
+use ambient_api::prelude::*;
+
+#[main]
+pub async fn main() {
+    Entity::new()
+        .with_merge(make_perspective_infinite_reverse_camera())
+        .with(aspect_ratio_from_window(), EntityId::resources())
+        .with_default(main_scene())
+        .with(translation(), vec3(2.0, 3.0, 4.0))
+        .with(lookat_target(), vec3(0.0, 0.0, 0.0))
+        .spawn();
+
+    let mut state = rand_pcg::Pcg64::seed_from_u64(0);
+    let dist = rand::distributions::Uniform::new_inclusive(0.0, 1.0);
+
+    let resolution: u32 = 32;
+    let mut temp_vertices = vec![];
+    let size = 8.0;
+    let half_size = 0.5 * size;
+    for y in 0..resolution {
+        for x in 0..resolution {
+            let x = (x as f32) / (resolution as f32);
+            let y = (y as f32) / (resolution as f32);
+            let z = 0.25 * dist.sample(&mut state);
+            let position = vec3(size * x - half_size, size * y - half_size, z);
+            temp_vertices.push(mesh::Vertex {
+                position,
+                normal: vec3(0.0, 0.0, 1.0),
+                tangent: vec3(1.0, 0.0, 0.0),
+                texcoord0: vec2(x, y),
+            });
+        }
+    }
+    let mut vertices = vec![];
+    let mut indices = vec![];
+    for y in 0..(resolution - 1) {
+        for x in 0..(resolution - 1) {
+            let i0 = x + y * resolution;
+            let i1 = (x + 1) + y * resolution;
+            let i2 = x + (y + 1) * resolution;
+            let i3 = (x + 1) + (y + 1) * resolution;
+            let mut v0 = temp_vertices[i0 as usize];
+            let mut v1 = temp_vertices[i1 as usize];
+            let mut v2 = temp_vertices[i2 as usize];
+            let mut v3 = temp_vertices[i3 as usize];
+            let n10 = (v1.position - v0.position).normalize();
+            let n20 = (v2.position - v0.position).normalize();
+            let n = n10.cross(n20).normalize();
+            v0.normal = n;
```

Comment:
Could be nice to have an option to calculate the normals automatically from the faces on the host side, but that's out of scope for this PR